pkg/reconciler/instances/connectivityproxy/command.go

```diff
@@ -37,48 +35,29 @@ type CommandActions struct {
 }
 
 func (a *CommandActions) InstallOnReleaseChange(context *service.ActionContext, app *appsv1.StatefulSet) error {
-	if app == nil || (app != nil && app.GetLabels() == nil) {
-		context.Logger.Debug("There is no valid Connectivity Proxy installed, invoking the installation")
-		return a.installOnCondition(context)
-	}
-
-	appName := app.Name
-	appRelease := app.GetLabels()[ReleaseLabelKey]
+	chartProvider := a.getChartProvider(context, app)
 
-	context.ChartProvider.WithFilter(func(manifest string) (string, error) {
-		unstructs, err := internalKubernetes.ToUnstructured([]byte(manifest), true)
-		if err != nil {
-			return "", errors.Wrapf(err, "while casting manifest to kubernetes unstructured")
-		}
+	err := a.install.Invoke(context.Context, chartProvider, context.Task, context.KubeClient)
+	if err != nil {
+		return errors.Wrap(err, "failed to invoke installation")
+	}
 
-		for _, unstruct := range unstructs {
-			if unstruct != nil && unstruct.GetName() == appName && unstruct.GetKind() == ConnectivityProxyKind {
-				if unstruct.GetLabels() == nil || unstruct.GetLabels()[ReleaseLabelKey] == "" {
-					return "", errors.Errorf("Connectivity Proxy stateful set does not have any release labels")
-				} else if unstruct.GetLabels()[ReleaseLabelKey] != appRelease {
-					context.Logger.Debug("Connectivity Proxy release has changed, the component will be upgraded")
-					return manifest, nil
-				} else {
-					context.Logger.Debug("Connectivity Proxy release did not change, skipping")
-					return "", nil
-				}
-			}
-		}
+	return nil
+}
 
-		context.Logger.Warn("Did not find the Connectivity Proxy stateful set, skipping")
-		return "", nil
-	})
+func (a *CommandActions) getChartProvider(context *service.ActionContext, app *appsv1.StatefulSet) chart.Provider {
```

Comment:
```suggestion
	if  app != nil && app.GetLabels() != nil {
		filters = append(filters,
			rendering.NewFilterByRelease(context.Logger, app.Name, app.GetLabels()[rendering.ReleaseLabelKey]),
			rendering.NewFilterByAnnotation("skip"))
	}
```

Reply:
Annotation filter makes sense only in case of upgrade

Reply:
Done